### ISSUE TITLE
MS disc: Last audio track cut short, data are partly inaccessible

### ISSUE THREAD

**Issue by Feathered-Serpent** [superg/redumper#46]
Hey there,

found your program and thought to give it a try on some multi session discs, which can't be dumped with DIC. And obviously also not with redumper. So far only CloneCD was able to dump them.

The first case is a CD where redumper correctly finds the 14 audio tracks in session 1, but cuts short the last audio track from 3:15 down to 1:30 playtime.

Session 2 with some data inside, seemingly incorrectly starts at LBA 211277, whereas on the CloneCD image, it starts at 219182:
![grafik](https://github.com/superg/redumper/assets/72162369/0cfdc9c7-2398-4a57-b675-1e0a86d57fca)

The ISO and the Joilet file system are marked as damaged from IsoBuster on the redumper image:
![grafik](https://github.com/superg/redumper/assets/72162369/634668f4-f4d1-43a3-bc6d-a5df370aba5c)

There is no data inside:
![grafik](https://github.com/superg/redumper/assets/72162369/bc78ff2c-b6ec-4ca6-b175-68f3b4dec055)

Whereas on the CloneCD ISO there is:
![grafik](https://github.com/superg/redumper/assets/72162369/e325a00d-5ae9-414a-a709-e1679c231cb2)

I'm not sure, what all I should provide, so I start with the log file from redumper:
[Original Motion Picture Score.log](https://github.com/superg/redumper/files/11545015/Original.Motion.Picture.Score.log)

Not sure if this would be of any help, but I copied the (german) log from CloneCD when I dumped the CD:
[POKEMON.log](https://github.com/superg/redumper/files/11545018/POKEMON.log)

**Comment by tjanas** [comment]
You probably need to modify your cuesheet to make the redumper or DIC image parse correctly in isobuster.

**Comment by Feathered-Serpent** [comment]
That indeed lets me extract the last audio track fully and I also can see the files (and extract them) in ISO and Joilet.

Btw. DIC doesn't dump the disc fully, but stops with an error close to the end. https://github.com/saramibreak/DiscImageCreator/issues/70#issuecomment-1003989049

Other question: redumper created a file named "Original Motion Picture Score (Track 00.2)". What is track 00.2? It's not linked in the cue sheet. The file consists nearly only out of 0-bytes.

**Comment by superg** [comment]
@Feathered-Serpent this is super interesting. From the log I see that there is TOC/subchannel mismatch but in addition to that it looks like there is some offset shifting going on. Will you be able to share the dump files for my investigation?

**Comment by superg** [comment]
also, 00.2 file is partial lead-in from the 2nd disc session. This file is created only if there is some non zero data there.

**Comment by superg** [comment]
Ah, wait, so that was IsoBuster problem?

**Comment by Feathered-Serpent** [comment]
I wouldn't say it's Isobuster problem, because with the cue sheet I cannot open any files when mounting the files with ImgDrive, and also when using ImgBurn to burn the cue disc onto a real disc I get tons of miscompares:
![grafik](https://github.com/superg/redumper/assets/72162369/41b2b558-fb80-4d84-b41c-8d5816642f28)

So I'm not sure if the cue should actually been able to be used to make a copy of the disc. 

I hope, 7z is not a problem: https://dragonswatch.silentstrider.net/sharing/OWn2quBqP that link expires in a week.

**Comment by superg** [comment]
I'll check the dump files tonight, thanks for uploading that.

Overall, there is a fundamental issue in how cue sheet (mis)represent multisession information.
Redumper creates multisession tags based on redump.org standards. Redump.org standards in turn were designed years ago and that some tags meaning contradict with what was initially meant by IsoBuster developer. There are a lot of details on that in DiscImageCreator issue tracker.

 TL;DR, I don't know any way on how can you burn multisession cue. What we do at redump.org, we cannot change our tags at this point because it will dramatically affect our cue sheets.

**Comment by Feathered-Serpent** [comment]
Ultimately, I do have my backup of the Audio/Enhanced CD made with CloneCD.

Opening such issues was just my thought, that I could try to eventually show problems, which could arrise on other discs as well. In case, there is something, that can be fixed.

So I would leave it up to you, if you follow up with the dump or not :) after your check that is.

**Comment by superg** [comment]
I checked your dump, I see no issues there, everything was dumped correctly. 
As I've explained before, I don't know any way to burn multisession CUE for a track split. First of all, multitrack dump of data disc is largely an invention of redump.org. Before that, dumping software was generating a single .img file for a data disc dump (CloneCD for instance). You will be able to burn your dump if you manually combine all tracks into one, such a combine will match 1:1 with CloneCD img file by the way.

If you're curious, here is the thread with more information on IsoBuster and multisession tags that we use: https://github.com/saramibreak/DiscImageCreator/issues/121

Some IDE-USB remarks: I own a number of drives that doesn't work with any USB adapters, 4012 is one of such drives I think..
There is some incompatibility which I have yet to understand and all such drives work if connected to native IDE or IDE-PCI card.

**Comment by Feathered-Serpent** [comment]
Thank you for the time to investigate the dump. If removing the leadout is anything that is needed to make the image work with Isobuster, then this shall be enough for me, as I have reached the goal to have a backup, I could work with from my NAS, in case the original disc ever can't be read anymore.

@IDE-USB: I suppose there are some IDE commands, that these adapters might not understand. Sounds strange, that it is this difficult to convert commands from such an old interface!

Case closed :)
